examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlClient.java

```diff
@@ -0,0 +1,237 @@
+import java.util.concurrent.TimeUnit;
+
+public class ManualFlowControlClient {
+  public static void main(String[] args) throws InterruptedException {
+    final ExecutorService pool = Executors.newCachedThreadPool();
+    final CountDownLatch done = new CountDownLatch(1);
+
+    // Create a channel and a stub
+    ManagedChannel channel = ManagedChannelBuilder
+        .forAddress("localhost", 50051)
+        .usePlaintext(true)
+        .build();
+    StreamingGreeterGrpc.StreamingGreeterStub stub = StreamingGreeterGrpc.newStub(channel);
+
+    // When using manual flow-control and back-pressure on the client, the ClientResponseObserver handles both
+    // request and response streams.
+    ClientResponseObserver<HelloRequest, HelloReply> clientResponseObserver =
+        new ClientResponseObserver<HelloRequest, HelloReply>() {
+
+          ClientCallStreamObserver<HelloRequest> requestStream;
+
+          @Override
+          public void beforeStart(final ClientCallStreamObserver<HelloRequest> requestStream) {
+            this.requestStream = requestStream;
+            // Set up manual flow control for the response stream. It feels backwards to configure the response
+            // stream's flow control using the request stream's observer, but this is the way it is.
+            requestStream.disableAutoInboundFlowControl();
+
+            // Set up a back-pressure-aware producer for the request stream. The onReadyHandler will be invoked
+            // when the consuming side has enough buffer space to receive more messages.
+            //
+            // Note: the onReadyHandler is invoked by gRPC's internal thread pool. You can't block in this in
+            // method or deadlocks can occur.
+            requestStream.setOnReadyHandler(new Runnable() {
+              // An iterator is used so we can pause and resume iteration of the request data.
+              Iterator<String> iterator = names().iterator();
+
+              @Override
+              public void run() {
+                // Start generating values from where we left off on a non-gRPC thread.
+                pool.execute(new Runnable() {
+                  @Override
+                  public void run() {
+                    // requestStream.isReady() will go false when the consuming side runs out of buffer space and
+                    // signals to slow down with back-pressure.
+                    while (requestStream.isReady()) {
+                      if (iterator.hasNext()) {
+                        // Simulate doing some work to generate the next value
+                        try {
+                          Thread.sleep(100);
+                        } catch (InterruptedException e) {
+                          e.printStackTrace();
+                        }
+
+                        // Send more messages if there are more messages to send.
+                        String name = iterator.next();
+                        System.out.println("--> " + name);
```

Comment:
nit: we have been using `java.util.logging.Logger` to print UI messages like these. I know it's a bit arbitrary :) but let's follow the convention here as well.